Apps/APN/App.js

```diff
@@ -0,0 +1,208 @@
+import Dashboard from './components/Dashboard';
+import CustomDataScreen from './components/CustomDataScreen';
+import SettingsScreen from './components/SettingsScreen'
+import Env from "./env";
+import CioManager from './manager/CioManager';
+import CioKeyValueStorage from './manager/KeyValueStorage';
+import Deeplinks from './components/Deeplink';
+
+const Stack = createNativeStackNavigator();
+
+export default function App() {
+
+  const [firstScreen, setFirstScreen] = useState(undefined)
+  const [loading, setLoading] = useState(true);
+const [isScreenTrackEnabled, setIsScreenTrackEnabled] = useState(null)
+const [isDeviceAttrTrackEnabled, setIsDeviceAttrTrackEnabled] = useState(null)
+const [isDebugModeEnabled, setIsDebugModeEnabled] = useState(null)
+const [bgDelayValue, setBgDelayValue] = useState(null)
+const [bgTasksValue, setBgTasksValue] = useState(null)
+const [trackingUrl, setTrackingUrl] = useState(null)
+
+  useEffect(() => {
+    (async () => {
+      const keyStorageObj = new CioKeyValueStorage()
+      const status = await keyStorageObj.getLoginStatus()
+      setLoading(false)
+      if (JSON.parse(status)) {
+        setFirstScreen("Dashboard")
+        return
+      }
+      setFirstScreen("Login")
+      
+    })();
+  }, [])
+
+  // Automatic screen tracking
+  const navigationRef = useNavigationContainerRef();
+  const routeNameRef = useRef();
+  const config = {
+    screens: {
+      Deeplinks: 'deeplink',
+    },
+  };
+  const linking = {
+    prefixes: ['amiapp://'],
+    config
+  };
+
+  useEffect(() => {
+    fetchConfigsOrSetDefault()
+  }, [])
+
+  useEffect(() => {
+    if(isDeviceAttrTrackEnabled !== null && isScreenTrackEnabled !== null && isDebugModeEnabled != null && bgDelayValue !== null && bgTasksValue !== null)
+    initialiseCioPackage()
+  }, [isDeviceAttrTrackEnabled, isScreenTrackEnabled, isDebugModeEnabled, bgDelayValue, bgTasksValue])
+
+  const fetchConfigsOrSetDefault = async() => {
+    const keyStorageObj = new CioKeyValueStorage()
+    const bgDelayValue  = await keyStorageObj.getBGQSecondsDelay()
+    const bgTasksValue = await keyStorageObj.getBGQMinTasksInQueue()
+    const screenTrackValue = await keyStorageObj.getScreenTrack()
+    const deviceAttrValue = await keyStorageObj.getDeviceAttributesTrack()
+    const debugModeValue = await keyStorageObj.getDebugModeConfig()
+    const trackUrl = await keyStorageObj.getTrackingUrl()
+
+    if (screenTrackValue === null) {
+      await keyStorageObj.saveScreenTrack(true)
+    }
+    if (bgDelayValue === null) {
+      await keyStorageObj.saveBGQSecondsDelay('30')
+    }
+    if (bgTasksValue === null) {
+      await keyStorageObj.saveBGQMinTasksInQueue('10')
+    }
+    if (deviceAttrValue === null) {
+      await keyStorageObj.saveDeviceAttributesTrack(true)
+    }
+    if (debugModeValue === null) {
+      await keyStorageObj.saveDebugModeConfig(true)
+    }
```

Comment:
This code looks like it is setting default values for the app's CIO SDK settings. 

If this is true, what is the reasoning for it? I don't think we need this code in here because the CIO SDK provides it's own default configuration values. The sample app shouldn't have it's own set of default values, too.

Reply:
👍 to this, we could reuse CIO SDK defaults here.

Reply:
We need the default code to set in the Settings screen, this is why you see that in case the values in local storage are null, I am setting the values to default to let the user know that these are the default values of the properties.

Reply:
The values being set in the code is currently equal to the SDK's default values, correct. 

What if in the future the SDK changes it's default value for BQ min number of tasks to 5? This code will become out-of-sync with the SDK's default value. 

I believe Shahroz and I are suggesting that the sample app only attempts to modify the SDK's configuration when the user of the app sets a new value in the settings app. If the user of the sample app opens the settings screen and does not modify, for example, the BQ min number of tasks then the sample app should continue to use the SDK's default configuration value. 

It looks to me like the code here is doing: If the user of the sample app opens the settings screen and does not modify, for example, the BQ min number of tasks then the sample app modifies the SDK's default configuration value to 10.